shell.nix

```diff
@@ -0,0 +1,34 @@
+{ date ? "2020-09-28" }:
+let
+  # To update nixpkgs or mozilla overlay pin:
+  #   1. Update `url` to the desired rev
+  #   2. Determine required sha256 by running `nix-prefetch-url --unpack <URL>`
+  pkgs = import (fetchTarball {
+   url = "https://github.com/NixOS/nixpkgs/archive/0cfe5377e8993052f9b0dd56d058f8008af45bd9.tar.gz";
+   sha256 = "0i3ybddi2mrlaz3di3svdpgy93zwmdglpywih4s9rd3wj865gzn1";
+  });
+  moz-overlay = import (fetchTarball {
+    url = "https://github.com/mozilla/nixpkgs-mozilla/archive/9f3df31ebb32ac97e385e07edd954fbf027ca242.tar.gz";
+    sha256 = "1swds7y664m04m1grjhypvslxzzwhiby4xqlc0aml62gbyi88jya";
+  });
+in
+with pkgs { overlays = [ moz-overlay ]; };
+mkShell {
+  buildInputs = [
+    (rustChannelOf { inherit date; channel = "nightly"; }).rust
+
+    pkgconfig
+    cmake
+    perl
+    go
+    sqlite
+    protobuf
+    openssl
+    valgrind
```

Comment:
```suggestion
    valgrind
    kustomize
    k9s
    kubeprompt
    dive
    stern
    kube3d
```
A selection of what we use and what might be useful...
<details>
<summary>Our full devshell entrypoint for reference</summary>

```nix
{ src
, preCommitExcludes ? []
, repoSpecificShellHook ? ""
}:
let
  nixpkgs-src = import ./nixpkgs-src.nix;
  overlay = import ./overlays/overlay.nix;
  pkgs = import nixpkgs-src {
    config = {};
    overlays = [ overlay ];
  };

  pre-commit = (
    import ./pre-commit.nix {
      inherit pkgs src;
      excludes = preCommitExcludes;
    }
  );
  devshell = pkgs.callPackage ( import ./devshell-src.nix ) { };

  # Overlays
  customTools = with pkgs; [
    j2cli
    commitizen
    container-structure-test
    kube-score
    skaffold-nix-builder
  ];

  nixTooling = with pkgs; [
    morph # gitops for nix (better UX, stateless)
    nix-diff # explains why two Nix derivations differ
    nix # nix package manager -- tagged version
    nixops # gitops for nix (stateful)
    nixos-container # run nixos in a systemd nspawn container (don't confuse with OCI compliant containers)
    nixpkgs-fmt # format nix files canonically
    nix-prefetch # determine the hash of a fixed-output derivation (TOFU)
    nix-top # help users figure out what's building
    nix-tree # interactively browse a Nix store paths dependencies
  ];

  gitAndRepoTooling = with pkgs; [
    bat # beat cat
    cfssl # generate certificates
    drone-cli # run CI locally
    fd # faster than find
    fzf # fuzzy find
    git
    gitAndTools.delta # better cli diffs
    gitAndTools.git-filter-repo # rewrite git history (eg extract subfolders)
    git-town # branching workflows
    hostctl # manage /etc/hosts file with style
    just # task runner
    kubeprompt # isolate kubconfigs in a shell
    nodePackages.prettier # auto format lots of things
    ripgrep # rest in peace grep!
    scc # calculate project cost / time COCOMO model
    shfmt # format shell files
    yq-go # yaml / json cli manipulator, akin to jq
  ];

  containerTooling = with pkgs; [
    dive
    k9s
    kpt
    kube3d
    kubectl
    kubernetes-helm
    kubeval
    kustomize
    lazydocker
    podman
    skaffold
    stern
  ];

in
devshell {
  name = "${baseNameOf src}";
  packages = nixTooling ++ gitAndRepoTooling ++ containerTooling ++ customTools;
  env = {
    # Fix the nixpkgs version to devshell's version
    # See: https://github.com/numtide/devshell/issues/26
    # NIX_PATH = "nixpkgs=${pkgs.path}";

    # common kube settings
    KUBECTL_EXTERNAL_DIFF = "delta";

    # common k9s settings
    K9SCONFIG = "${toString ./k9s.d}";

		# Skaffold environment for K8sApp based dev environments (not relevant in neptun)
		# TODO: secure MTL & CFX regitries
		SKAFFOLD_INSECURE_REGISTRY = "registry.quantumops.rocks:5000";
  };
  bash = {

    extra = ''
    playground_zone_kubekonfig="$HOME/.kube/git.p1.rocks/gitops/DevShell/playground-zone"
    k3d_playground_kubeconfig="$HOME/.kube/git.p1.rocks/gitops/DevShell/k3d-playground"
    export KUBECONFIG="$KUBECONFIG:$HOME/.kube/config:$playground_zone_kubekonfig:$k3d_playground_kubeconfig";
    # git town settings
    git config git-town.code-hosting-driver gitea
    git config git-town.code-hosting-origin-hostname git.p1.rocks
    git config git-town.main-branch-name master

    # git delta settings
    git config core.pager delta
    git config interactive.diffFilter "delta --color-only"
    git config diff.colorMoved default
    git config delta.navigate true
    git config delta.side-by-side true
    git config delta.line-numbers true
    git config delta.decorations true
    git config delta.whitespace-error-style "22 reverse"
    ${repoSpecificShellHook}
    ${pre-commit.shellHook}

    # zsh prompt configuration
    export IN_WHICH_NIX_SHELL="$(basename $(pwd))"
    export KUBEPROMPT_FORMAT='{{if .Enabled}}%B%F{blue}%{⎈ %2G%}%f➜%{ %}%F{yellow}{{.Ctx}}%f|%F{magenta}{{.Ns}}%f%b{{end}}'
    export RPROMPT='$(kubeprompt -f default)'

    # Fix the nixpkgs version to devshell's version
    export NIX_PATH="nixpkgs=${pkgs.path}"
    '';

    interactive = ''

    '';
  };
}

```

</details>